backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java

```diff
@@ -0,0 +1,244 @@
+package ee.tlu.forum.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import ee.tlu.forum.model.Post;
+import ee.tlu.forum.model.Thread;
+import ee.tlu.forum.model.User;
+import ee.tlu.forum.model.input.AddNewPostInput;
+import ee.tlu.forum.service.PostService;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.RequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultHandlers;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest
+class PostControllerTest {
+
+    public MockMvc mockMvc;
+
+    @Autowired
+    PostController postController;
+
+    @MockBean
+    public PostService postService;
+
+    @BeforeEach
+    public void setUp() {
+        mockMvc = MockMvcBuilders.standaloneSetup(postController).build();
+    }
+
+    @Test
+    @DisplayName("Returns a list of posts at - GET /api/threads")
```

Comment:
Feeling like a bit of a broken record here but should we, perhaps, test a list with at least two posts?

Reply:
Sure, but that'll have to wait until the next week as I'm not available - unless you'd like to do it!

Reply:
It was just something to note for the future :)